spec/javascript/controllers/table_controller_spec.js

```diff
@@ -0,0 +1,58 @@
+import { Application } from 'stimulus';
+import TableController from 'table_controller.js';
+
+describe('TableController', () => {
+
+  beforeAll(() => registerController());
+
+  const setBody = () => {
+    document.body.innerHTML = `
+      <div data-controller="table">
+        <div class="markdown">
+          <table>
+            <thead>
+              <tr>
+                <th class="empty-col"><!-- empty --></th>
+                <th class="col">Col 1</th>
+                <th class="col">Col 2</th>
+                <th class="col-with-scope" scope="row">Col 3</th>
+              </tr>
+            </thead>
+            <tbody>
+              <tr>
+                <td>Cell 1</td>
+                <td>Cell 2</td>
+                <td>Cell 3</td>
+                <td>Cell 4</td>
+              </tr>
+            </tbody>
+          </table>
+        </div>
+      </div>
+    `;
+  }
+
+  const registerController = () => {
+    const application = Application.start();
+    application.register('table', TableController);
+  }
+
+  beforeEach(() => {
+    setBody();
+  });
+
+  it('adds a col scope to headings', () => {
+    const cols = [...document.querySelectorAll('.col')]
+    expect(cols.map(c => c.scope)).toEqual(['col', 'col']);
+  })
+
+  it('does not add a scope to empty headings', () => {
+    const emptyCol = document.querySelectorAll('.empty-col')
+    expect(emptyCol.scope).toBeUndefined();
```

Comment:
I think this test will always fail? `querySelectorAll` returns a nodeList.

```js
    const emptyCol = document.querySelector('.empty-col')
    expect(emptyCol.scope).toBe(false) // or something, I think it will be an empty string which is false in JS
```

Reply:
Whoops, yep well spotted